scripts/e2e-tests.sh

```diff
@@ -1,14 +1,18 @@
 #!/bin/bash
 set -e
 
-TEST_ARG="${1:-*}"
-
-MYDIR=$(realpath "$(dirname "$0")")
-cd "$MYDIR"
-cd ..
+# Move to project root
+cd "$(dirname "$(realpath "$0")")/.."
 
+# Build the project
 cargo build --release --target wasm32-unknown-unknown -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort
 
+# Set RPC_URL environment variable
 export RPC_URL=http://localhost:8547
 
-cargo test --features std,e2e --test "$TEST_ARG"
+# Run tests based on arguments
+if [[ $# -eq 0 ]]; then
+    cargo test --features std,e2e --test "*"
+else
+    cargo test --features std,e2e "$@"
+fi
```

Comment:
The script can now be called in the following ways:

- if no arguments are passed, the behavior is the same as before
- otherwise, pass whatever arguments are present to `cargo test` directly

We can now easily filter for even unit tests with this script. 

This should cover most of our testing needs.

NOTE: feel free to suggest or add any additional feature you think might be useful.

Reply:
An other option is to omit the if-statement, and just call `cargo test --features std,e2e "$@"`, passing whatever arguments to the command. In our e2e workflow, we'd then call the script like `./scripts/e2e-tests.sh --test "*"`.

At this point it's really a matter of taste

Reply:
Let's leave right now the current behaviour.

Reply:
Basically, it lets you call this script as if you were to call `cargo test`, meaning it enables you to do all the funky filtering stuff

Reply:
All of these are now supported

```bash
# run all tests (appends `--test "*"` as before)
./scripts/e2e-tests.sh
# pass arguments to `cargo test`
./scripts/e2e-tests.sh --test erc721
./scripts/e2e-tests.sh --no-run
# pass arguments both to `cargo test` and to test binary
./scripts/e2e-tests.sh --test erc721 -- support_interface
./scripts/e2e-tests.sh erc721 --no-run -- --nocapture
```

Reply:
I see.
But in this case it is impossible to invoke the command like `./scripts/e2e-tests.sh erc721 error_when_minting_token_id_twice`

If you would change it like this:
```
if [[ $# -eq 0 ]]; then
    cargo test --features std,e2e --test "*"
else
    cargo test --features std,e2e --test "$@"
fi
```
It can work.

Reply:
And we should spell it like `./scripts/e2e-tests.sh --test erc721 error_when_minting_token_id_twice`

Reply:
But actually what I like very much in your approach, is that if we spell `./scripts/e2e-tests.sh --help` it will print docs for cargo sub commands. So no need to bother ourselves with script's documentation

Reply:
The current implementation is the simplest one, with the most straightforward way to use.

Updating the script to support invocations like:
```bash
./scripts/e2e-tests.sh erc721 error_when_minting_token_id_twice
```
Would make it very complex if we also wanted to support passing other valid args/flags to `cargo test`, or to the test binary.

> But actually what I like very much in your approach, is that if we spell ./scripts/e2e-tests.sh --help it will print docs for cargo sub commands. So no need to bother ourselves with script's documentation

Can you clarify the last sentence?

Reply:
Let's keep you changes

Reply:
> Can you clarify the last sentence?
I mean, that we can print valid documentation for `e2e-test.sh` script, since we're just forwarding arguments to `cargo test` like:
```./scripts/e2e-tests.sh --help```

Reply:
Does the script then have your blessing? :smile: